### DISCUSSION TITLE
Guidance on Contributing to ROCm with MacBook M1 Pro

### DISCUSSION THREAD

**Discussion by charan-003** [ROCm/ROCm#4049] [Q&A]
Hi ROCm Community,
I use a MacBook M1 Pro, which I understand isn’t fully compatible with ROCm’s hardware requirements. However, I’m eager to contribute in any way I can. From my research, I assume I can work on:

Improving documentation.
Writing or enhancing unit tests.
Fixing beginner-friendly issues in repositories like MIOpen
Could you please confirm if these are good starting points, or suggest other ways I can contribute effectively without access to AMD GPUs?

I recently came across Docker containers and am curious if they might help me get started with ROCm development. I’m not entirely sure how to set them up or use them effectively. If Docker is a viable option, could someone guide me on how to proceed?

I’m excited to learn and contribute to this amazing project. Thank you in advance for your guidance!

**Comment by schung-amd** [comment]
Hi @charan-003, thanks for your interest! Any suggestions on documentation improvement are more than welcome. In general, as the ROCm stack is open source, you are also free to submit pull requests to improve the codebase. 

However, anything that requires actually using HIP/ROCm will not function natively on your hardware, as we do not currently support ARM-based systems. Docker containers will not solve this issue on their own. I suggest getting access to a system with an AMD GPU if you're interested in contributing to ROCm.

**Comment by charan-003** [comment]
@schung-amd Thank you for your response! I completely understand the hardware limitations for ARM-based systems. I’m eager to contribute to ROCm and was wondering if there are any online resources or virtual machines available that provide access to AMD GPUs for development purposes.

If such options aren’t available, I’m open to setting up a dedicated machine with compatible hardware. Could you provide some guidance on how best to get started? I plan to begin with the documentation but would greatly appreciate any tips or recommendations for a beginner interested in learning more about ROCm and working with AMD technologies. Thank you in advance for your support!

**Comment by schung-amd** [comment]
I'm not aware of any easy way to access ROCm-compatible systems at the moment, it's probably best to set up a dedicated system on your end to develop on. Unfortunately our current official GPU support (see: https://rocm.docs.amd.com/projects/install-on-linux/en/latest/reference/system-requirements.html) is sparse compared to CUDA, so this isn't exactly trivial, but some users have gotten ROCm working on many other GPUs than are listed here. I've personally had ROCm working on a 6800XT in the past, and I've gotten some workloads running on a 6400, but your mileage may vary.

Regarding where to start with ROCm, it depends on your area of interest, as the ROCm stack is involved in a lot of workflows. For basic GPGPU usage in general, we have a HIP programming guide: https://rocm.docs.amd.com/projects/HIP/en/latest/programming_guide.html. We were/are reworking parts of this, not sure what the status is. The basic principles are similar to CUDA, so you can refer to their programming guides as well. You can check out the ROCm examples (https://github.com/rocm/rocm-examples) as a way to verify that your ROCm installation is functional but also to get an idea of the basic things that can be done with HIP. On the machine learning side of things, ROCm backends are implemented in frameworks like PyTorch and TensorFlow, so you can jump right into ML workloads without needing to dive into the lower level elements beforehand. For specific ROCm components you can navigate to their repos and see if they have tests or examples available.

In terms of looking at issues, many issues end up being documentation or communication-related, so starting there can be useful. For deeper issues, it is always helpful to get reproducers, and many community members have also done their own debugging deep dives. In that regard I would suggest familiarizing yourself with debugging tools (such as rocgdb) and profiling tools (such as rocprof). Even if you can't address an issue from start to finish, we're appreciative of any and all contributions from community members.
